### PR TITLE
Add environment variable to select storage provider

### PR DESCRIPTION
**Pull request by minosimo**

### CODE REVIEW

**Comment by bermeitinger-b** [comment]
Could you also add the explanation of this variable to the Readme?

This should be tested.

File: signal-desktop.sh [hunk 1]
```diff
@@ -10,6 +10,12 @@ if (( ${SIGNAL_USE_WAYLAND:-0} )); then
     export ELECTRON_OZONE_PLATFORM_HINT="${ELECTRON_OZONE_PLATFORM_HINT:-auto}"
 fi
 
+SIGNAL_PASSWORD_STORE="${SIGNAL_PASSWORD_STORE:-basic}"
+
+EXTRA_ARGS+=(
+    "--password-store=${SIGNAL_PASSWORD_STORE}"
+)
+
```
Comment (xduugu) [inline signal-desktop.sh:18]: Given that `--password-store` is unconditionally added to `EXTRA_ARGS`, you could just replace line 3

```bash
EXTRA_ARGS=()
```

with

```bash
EXTRA_ARGS=(
     "--password-store=${SIGNAL_PASSWORD_STORE:-basic}"
)
```

**Comment by minosimo** [comment]
Valid options taken from here: https://www.electronjs.org/docs/latest/api/safe-storage

I'm assuming kwallet6 is a valid command line option even though it isn't explicitly listed.

**Review by bermeitinger-b — APPROVED** [review]

**Comment by bbhtt** [comment]
I think adding a warning message here is useful. Something like this:

```patch
diff --git a/signal-desktop.sh b/signal-desktop.sh
index 926e41d..ea590e8 100644
--- a/signal-desktop.sh
+++ b/signal-desktop.sh
@@ -1,5 +1,28 @@
 #!/bin/bash
 
+report_warning() {
+    read -r -d '|' MESSAGE <<EOF
+Signal is being launched with the <b>plaintext password store</b> by
+default due to database corruption bugs when using the encrypted backends.
+This will leave your keys <b>unencrypted</b> on disk.
+
+If you wish to experiment with the encrypted backend at the risk of 
+database corruption, set the environment variable
+<tt>SIGNAL_PASSWORD_STORE</tt> to gnome_libsecret, kwallet,
+kwallet5 or kwallet6 depending on your desktop environment using
+Flatseal.
+
+Press <b>Yes</b> to proceed with <b>plaintext password store</b> or
+<b>No</b> to <b>exit</b>. |
+EOF
+    zenity --question --no-wrap --default-cancel --icon-name=dialog-warning --title "Warning" --text "$MESSAGE"
+
+    if [ "$?" -eq "1" ]; then
+        echo "Debug: Abort as user pressed no"
+        exit 1
+    fi
+}
+
 EXTRA_ARGS=()
 
 declare -i SIGNAL_DISABLE_GPU="${SIGNAL_DISABLE_GPU:-0}"
@@ -30,4 +53,9 @@ echo "Debug: Will run signal with the following arguments:" "${EXTRA_ARGS[@]}"
 echo "Debug: Additionally, user gave: $*"
 
 export TMPDIR="${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}"
+
+for value in "${EXTRA_ARGS[@]}"; do
+    [[ "--password-store=basic" = "$value" ]] && report_warning
+done
+
 exec zypak-wrapper "/app/Signal/signal-desktop" "${EXTRA_ARGS[@]}" "$@"
```

**Comment by bermeitinger-b** [comment]
Thanks @bbhtt  for this script. I've boiled it down to a mere print of "Info: ".

**Comment by bbhtt** [comment]
The problem with `info: foo bar` on the command line is no one will notice it.

**Comment by minosimo** [comment]
While we work out the details of this MR, is there any reason to not merge #737 immediately?

**Comment by minosimo** [comment]
The warning in its current form is likely to confuse many users. The Flathub build has been using using the unencrypted store for a long time, and it isn't clear why the warning should appear now when nothing is changing. I don't think it will be easy to explain in the popup either.

Could we show the warning on new installs, but suppress it for existing ones?

I will also repeat my suggestion that we simply merge #737 immediately to prevent the bad update being pushed to more systems. The current version breaks both existing and new installs. Without a backup, there is no way to restore messages into Signal desktop once they have been rendered inaccessible.

**Comment by bermeitinger-b** [comment]
That's good!

I've reworded the message slightly and added input validation in the **minosimo-storage_env_variable** branch. Could you add this to this MR?

**Comment by minosimo** [comment]
bermeitinger-b's new MR is here: https://github.com/flathub/org.signal.Signal/tree/minosimo-storage_env_variable

**Pull request closed without merging**